spec/compare_spec.rb

```diff
@@ -6,15 +6,13 @@
     it "sorts lists recursively, including lists of hashes" do
       data = {a: [{b:[2]}, {b:[1]}]}
       sorted = EcsCompose::Compare.sort_recursively(data)
-      expect(data).to eq({a: [{b:[2]}, {b:[1]}]})
```

Comment:
This check is to make sure we don't destructively modify `data` when sorting it. We should probably call out this test explicitly so it's clear, not delete it.